### ISSUE TITLE
Freezing/lagging dragging after certain time

### ISSUE THREAD

**Issue by chuckdu21** [artginzburg/MiddleClick-Sonoma#111]
Hi,
I have your app for a long time now (and it's really helpful, thanks by the way), but i noticed after like few weeks of use (so no reboot of the app or mac) when i want to drag and drop it's lagging. It can be a tab in chrome, a file in finder, a track in rekordbox.

At the beginning i didn't know why my macbook was lagging like that but this morning i reboot the app and no lagging anymore after that. So the issue comes from your app.

Now, the question is why ?

**Comment by artginzburg** [comment]
@2mawi2 that is actually how we did it before. Then, if I remember right, @ris58h recommended that we eliminate that dirty hack and instead make the app recreate its listeners.

**Comment by artginzburg** [comment]
@2mawi2 thanks for the PR!
What do you think about enabling Automatic Reference Counting for the app? As far as I can tell, in theory it prevents memory leaks, and costs almost nothing in terms of CPU, Memory and Energy usage. I'm going to try that and profile the app a bit

**Comment by ris58h** [comment]
I've faced this issue too. What is the root cause BTW?

**Comment by artginzburg** [comment]
By chance, do any of you guys have ProMotion enabled on the display? This may be connected to #105 and #77

**Comment by ris58h** [comment]
I use MBP 14 M3 without any external display if it helps.

**Comment by slowan** [comment]
> By chance, do any of you guys have ProMotion enabled on the display (built into new MBPs and enabled by default)? This may be connected to [#105](https://github.com/artginzburg/MiddleClick-Sonoma/issues/105) and [#77](https://github.com/artginzburg/MiddleClick-Sonoma/issues/77)

I am using Macbook pro M1 Max with ProMotion enabled.

Occasionally, it adds quite long initial lag before scrolling happen and then it's a bit choppy.
Quit the app and start it again solves the problem.

What I noticed is that it probably happen after Mac is waked up from sleep (connected standby disabled in Windows language), but I can't say it for 100%, maybe it happens also after LCD is waked up (I have sleeping disabled when connected to adapter).

Note: I also use Mac Mouse Fix for standard mouse scrolling. But only MiddleClick restart helps with the lag and stuttering.

**Comment by utopiaeh** [comment]
The problem still persists. Any updates about this?

**Comment by JosephTico** [comment]
> The problem still persists. Any updates about this?

Looks like the fix has already been merged but a new release hasn't been made yet, you can compile the latest code yourself and it should be fixed

**Comment by artginzburg** [comment]
Should we dedupe #105 and #111 into #77?

**Comment by salman-ar-sar** [comment]
> Should we dedupe [#105](https://github.com/artginzburg/MiddleClick/issues/105) and [#111](https://github.com/artginzburg/MiddleClick/issues/111) into [#77](https://github.com/artginzburg/MiddleClick/issues/77)?

Yes, I think all of them seem to be the same issue. BTW, did #113 slow this issue?
